app/src/components/report/LocationPicker.tsx

```diff
@@ -0,0 +1,222 @@
+import { Form, Button } from 'react-bootstrap';
+import { isUndefined } from '@/utils';
+import { useLocationPicker } from '@/hooks';
+import MyModal, { ModalHandle } from './MyModal';
+
+const MAPBOX_ACCESS_TOKEN = process.env.NEXT_PUBLIC_MAPBOX_ACCESS_TOKEN;
+const DEFAULT_COORDINATES = { longitude: -95.844032, latitude: 36.966428, zoom: 3 };
+
+export type CoordinatesType = {
+    longitude: number;
+    latitude: number;
+};
+
+type ViewStateType = CoordinatesType & {
+    zoom: number;
+};
+
+type LocationSearchBarProps = {
+    placeMarker: (location: CoordinatesType) => void;
+    proximity?: CoordinatesType;
+};
+
+type LocationPickerProps = {
+    reportCoords: CoordinatesType | undefined;
+    initialMarkerState: boolean;
+    locationSubmitted: boolean;
+    updateReportCoords: (newCoords: CoordinatesType) => void;
+    fillOutAddress: (address: string) => void;
+};
+
+/** Save user reported location in local storage */
+export function submitLocation(reportCoords: CoordinatesType | undefined) {
+    if (typeof window === 'undefined' || isUndefined(reportCoords)) return;
+    localStorage.setItem('lastReportedLocation', JSON.stringify(reportCoords));
+}
+
+/** Location Search Bar */
+function LocationSearchBar({ placeMarker, proximity }: LocationSearchBarProps) {
+    const geocoder = new MapboxGeocoder({
+        accessToken: MAPBOX_ACCESS_TOKEN,
+        mapboxgl: mapboxgl,
+        marker: false,
+        placeholder: 'Where do you report it?',
+        countries: 'us',
+        proximity,
+    });
+
+    useControl(() => geocoder);
+
+    geocoder.on('result', (event) => {
+        const [longitude, latitude] = event.result.geometry.coordinates;
+        placeMarker({ longitude, latitude });
+    });
+
+    return null;
+}
+
+/** The map */
+export default function LocationPicker({
+    reportCoords,
+    updateReportCoords,
+    fillOutAddress,
+    initialMarkerState,
+    locationSubmitted,
+}: LocationPickerProps) {
+    /** Retrieve last user reported location for the view of the map upon loading page*/
+    function fetchUserLastLocation(): ViewStateType {
+        if (typeof window === 'undefined') return DEFAULT_COORDINATES;
+
+        if (isUndefined(reportCoords)) {
+            const lastReportedLocation = localStorage.getItem('lastReportedLocation');
+
+            if (lastReportedLocation) {
+                const { longitude, latitude } = JSON.parse(lastReportedLocation);
+                return { longitude, latitude, zoom: 10.5 };
+            } else {
+                return DEFAULT_COORDINATES;
+            }
+        } else {
+            return { ...reportCoords!, zoom: 10.5 };
+        }
+    }
+    // View Box of the map upon entering a page
+    const [viewState, setViewState] = useState<ViewStateType | {}>(fetchUserLastLocation);
+    // Disable the marker after user already submit location
+    const [draggableMarker, setDraggableMarker] = useState<boolean>(initialMarkerState);
```

Comment:
What does this variable mean? Could it be clearer as `isDraggableMarkerEnabled` or similar?